modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/KotlinClientCodegen.java

```diff
@@ -151,25 +241,97 @@ public void processOpts() {
             typeMapping.put("list", "kotlin.collections.List");
             additionalProperties.put("isList", true);
         }
+    }
+
+    @Override
+    public Map<String, Object> postProcessModels(Map<String, Object> objs) {
+
+        // workaround for: https://github.com/swagger-api/swagger-codegen/issues/4258
```

Comment:
Do we still need this workaround? 

I think the issue has been fixed in OpenAPI Generator

Reply:
Not sure. I couldn't find a reference to the issue in OpenAPI Generator (and I couldn't remember in what scenario it was necessary). I'll run through some tests and get back to you.

Reply:
Yes, please. If it's still an issue, we need to track it and get it fixed.

Reply:
Confirmed fixed. Unnecessary code removed.

Reply:
I've also included a fix for [this](https://github.com/OpenAPITools/openapi-generator/issues/3917) issue. Happy to submit as a separate pull request if desired.

https://github.com/OpenAPITools/openapi-generator/issues/3917

Reply:
Prefer a separate PR

Reply:
Separate pull request created:
https://github.com/OpenAPITools/openapi-generator/pull/3928